app/controllers/tasks_controller.rb

```diff
@@ -1,80 +1,133 @@
 class TasksController < ApplicationController
 
   def index
-    @all_tasks = Task.all   # NOTE TO SELF: "task" is actually a Task::ActiveRecord_Relation object, but responds to Array methods
-    completed_instance_var
+    @all_tasks = Task.all
+      # NOTE TO SELF: "@all_tasks" is actually a Task::ActiveRecord_Relation object, but responds to array methods
 
-    render :index   # for clarity, you don't actually need this in this case
+    render :index
+      # NOTE TO SELF: this is for clarity, you don't actually need this in this case
+  end
+
+  def completed_status
+    @task = Task.find(params[:id])
+    if @task[:completed_at].nil?
+      Task.completed(params[:id])
+    else
+      Task.not_completed(params[:id])
+    end
+
+    redirect_to "/"
   end
 
   def show
-    task_instance_vars
-    @visibility = "invisible"
+    task = Task.find(params[:id])
+
+    @task_id =              task[:id]
+    @task_name =            task[:name]
+    @task_description =     task[:description]
+    @task_status =          task[:completed_at].nil? ? "NEED TO PLAY" : "FINISHED"
+    @task_completion_date = task[:completed_at].nil? ? "N/A" : date_format(task[:completed_at])
+    @task_created_at =      date_format(task[:created_at])
+    @task_updated_at =      date_format(task[:updated_at])
+    @task_recommended_id =  task.person.id
+    @task_recommended_by =  task.person.name
+
+    @button_visibility = "visible"
+    @delete_visibility = "invisible"
 
     render :task
+      # NOTE TO SELF: "render 'task'" also would work
   end
 
   def show_before_delete
     task_instance_vars
-    @visibility = "visible"
+    @button_visibility = "invisible"
+    @delete_visibility = "visible"
 
     render :task
   end
 
   def new
     @task = Task.new
+    @person = Person.new
```

Comment:
Do you need this instance variable here? Do you create a new instance of a person when you create a new instance of a task?